### PR TITLE
decorate person class with decorators

### PR DESCRIPTION
**Pull request by Jaymelfah**
#### In this branch, I used two decorators to capitalize and trim names following the decorator design pattern.

> Created a class called Nameable
> Created a base decorator
> Created a capitalize decorator
> Created a trim decorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
-class Person
+require './nameable'
+require './capitalize_decorator'
+require './trimmer_decorator'
```
Comment (rloterh) [inline person.rb]: - [ ] OPTIONAL
By convention, it's a good practice to use `require_relative` classes in the same source and use `require` for external modules or packages. Kindly ensure to improve upon this aspect.

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (rloterh) [inline base_decorator.rb:6]: - [ ] OPTIONAL
It's a best practice to mark `nameable` with `attr_accessor`, to give an indication that we intend to both _read_ and write this variable. Let's endeavor to improve upon the same.

File: base_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
```
Comment (rloterh) [inline base_decorator.rb]: - [ ] Overridden `correct_name` method is not properly implemented in _BaseDecorator_ class . `correct_name` here should return the result of the `correct_name` of `@namable`. This should be similar to:

```ruby
class Decorator < Nameable
    attr_accessor :nameable
  
  ...

    def correct_name
      @nameable.correct_name
    end
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Jay, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Awesome job with decorator implementation so far!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added.
- MIT license link in your ReadMe is broken, kindly endeavor to fix it, and remove the guide note below it.

Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Jaymelfah 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**